Update vimrc

zaratan opened the pull request.

Commit:
Update vimrc

Mate2xo commented on vimrc:
```diff
@@ -5,6 +5,10 @@ if &compatible
 endif
 
+" You want utf-8 by default
+set encoding=utf-8
```
This seems to be already set by vim-sensible plugin

Mate2xo commented on vimrc:
```diff
@@ -71,11 +82,40 @@ nmap <C-p> :FZF<CR>
+set ttymouse=sgr
+
+" Try to autoindent
+set autoindent
```
already set by vim-sensible

Mate2xo commented on vimrc:
```diff
@@ -71,11 +82,40 @@ nmap <C-p> :FZF<CR>
+" Try to autoindent
+set autoindent
+set smartindent
+set smarttab
```
already set by vim-sensible

Mate2xo commented on vimrc:
```diff
@@ -71,11 +82,40 @@ nmap <C-p> :FZF<CR>
+
+" Better search
+set hls is
+set incsearch
```
already set by vim-sensible

Mate2xo commented on vimrc:
```diff
@@ -19,6 +23,13 @@ colorscheme codedark
+set rtp+=/usr/local/opt/fzf
+nmap ; :Buffers<CR>
+nmap <Leader>e :Files<CR>
+nmap <Leader>r :Tags<CR>
```
Is this a common shortcut? I'm tempted to use `<Leader>t` (more semantic)

Mate2xo committed:
add Fzf config, folding, system clipboard

also search hilight, smart indent, no line limit size, extended mouse support, extended color setup for tmux
